src/main/resources/db/changelog/20210127105630-migrate-product-hierarchy.xml

```diff
@@ -24,25 +124,109 @@
         </comment>
 
         <sql>
-            UPDATE cp2_products prod JOIN cp_pool pool ON pool.product_uuid = prod.uuid
-            SET prod.derived_product_uuid = pool.derived_product_uuid
-        </sql>
-    </changeSet>
+            -- We cannot create some of these as temporary tables, as MySQL/MariaDB cannot use the same
+            -- temporary table multiple times in a given query (which we do often)
+            -- Note the table creations use windowing functions, which require MySQL 8.0+ or MariaDB 10.0+
+            DROP TABLE IF EXISTS tmp_product_map;
+            CREATE TABLE tmp_product_map (
+                product_id VARCHAR(32) PRIMARY KEY,
+                product_uuid VARCHAR(32) NOT NULL UNIQUE
+            );
 
-    <changeSet id="20210127105630-2" author="crog">
-        <comment>
-            Migrates provided products and derived provided products from pools to products
-        </comment>
+            DROP TABLE IF EXISTS tmp_product_pools;
+            CREATE TEMPORARY TABLE tmp_product_pools (
+                pool_id VARCHAR(32) PRIMARY KEY,
+                product_id VARCHAR(32) NOT NULL UNIQUE,
+                derived_product_id VARCHAR(32),
```

Comment:
It might be allowed at the DB level, but no such pools exist in prod, and we have the \@NotNull annotation on our base product within CP -- it shouldn't be a thing without DB fiddling.